src/context/dataContext.js

```diff
@@ -0,0 +1,41 @@
+import { useContext, useReducer, useEffect, createContext } from "react";
+import { reducer } from "../reducer/reducer";
+import { getCategories, getVideos } from "../services/dataService";
+
+const initialState = {
+  videos: [],
+  categories: [],
+  category: "All",
+};
+const DataContext = createContext();
+
+const DataProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  useEffect(
+    () =>
+      (async () => {
+        const newVideos = await getVideos();
```

Comment:
Instead of running the getVideos and getCategories functions separately , you can run them in a single line itself using Promise.all
Here is a small demo:
```js
 const [response1 , response2 ] = await Promise.all( [ getVideos() , getCategories() ] ) 

```

I don't know about how to call functions within it but maybe we can try it out